www/js/site.js

```diff
@@ -1193,3 +1193,26 @@ var getScrollbarWidth = function() {
     };
   }
 })();
+
+// handle script file uploads
+const uploadInput = document.getElementById("script_file");
+if (uploadInput) {
+  uploadInput.addEventListener(
+    "change",
+    () => {
+      const file = uploadInput.files[0];
+      const reader = new FileReader();
+      reader.addEventListener(
+        "load",
+        () => {
+          document.getElementById("enter_script").value = reader.result;
```

Comment:
How do we prevent this from being used maliciously?

Reply:
Thanks! As discussed offline, we should be fine (famous last words) :)